app/src/main/java/org/p2p/wallet/history/ui/history/HistoryFragment.kt

```diff
@@ -1,19 +1,65 @@
 package org.p2p.wallet.history.ui.history
 
+import android.os.Bundle
+import android.view.View
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
 import org.koin.android.ext.android.inject
 import org.p2p.wallet.R
 import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.common.ui.recycler.EndlessScrollListener
+import org.p2p.wallet.common.ui.recycler.PagingState
 import org.p2p.wallet.databinding.FragmentHistoryBinding
+import org.p2p.wallet.history.model.HistoryTransaction
+import org.p2p.wallet.history.ui.token.adapter.HistoryAdapter
+import org.p2p.wallet.utils.unsafeLazy
 import org.p2p.wallet.utils.viewbinding.viewBinding
+import timber.log.Timber
 
 class HistoryFragment :
     BaseMvpFragment<HistoryContract.View, HistoryContract.Presenter>(R.layout.fragment_history),
     HistoryContract.View {
 
-    override val presenter: HistoryContract.Presenter by inject()
-
     companion object {
         fun create() = HistoryFragment()
     }
+
+    override val presenter: HistoryContract.Presenter by inject()
     private val binding: FragmentHistoryBinding by viewBinding()
+    private val adapter: HistoryAdapter by unsafeLazy {
+        HistoryAdapter(
+            onTransactionClicked = {},
+            onRetryClicked = {}
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        Timber.tag("________").d("onViewCreate")
```

Comment:
strange tag
maybe we do not need it ?)

Reply:
yeah i just test fragment creation while i switch tabs